packages/@ember/-internals/glimmer/tests/integration/helpers/closure-action-test.js

```diff
@@ -250,14 +250,10 @@ moduleFor(
       }, /An action could not be made for `.*` in .*\. Please confirm that you are using either a quoted action name \(i\.e\. `\(action '.*'\)`\) or a function available in .*\./);
     }
 
-    ['@test [#12718] a nice error is shown when a bound action function is undefined and it is passed as attrs.foo']() {
-      expectDeprecation(
-        "Using {{attrs}} to reference named arguments has been deprecated. {{attrs.external-action}} should be updated to {{@external-action}}. ('my-app/templates/components/inner-component.hbs' @ L1:C43) "
-      );
-
+    ['@test [#12718] a nice error is shown when a bound action function is undefined and it is passed as this.attrs.foo']() {
```

Comment:
```suggestion
    ['@test [#12718] a nice error is shown when a bound action function is undefined and it is passed as @foo']() {
```

Reply:
would we not want both? or is that overkill?

Reply:
`this.attrs` should have nothing "special" about it in 4.0. It didn't in 3.x either, except for the deprecation warning.

The question here is: Is the test asserting when a the actions is passed via an internal property of the component class (`this.any.thing.at.all` including `this.attrs.foo`) or read from external arguments (`@foo`)?

In my reading the test appears more targeted toward an external argument. The variable for the action is named `external-action` for example.